File: python/templates/rng.py
# python/templates/rng.py
import random


def new_rng(seed: int) -> random.Random:
    return random.Random(seed)


def randint(rng: random.Random, n: int) -> int:
    return rng.randint(0, n - 1)


def shuffle(rng: random.Random, items: list) -> None:
    rng.shuffle(items)


if __name__ == "__main__":
    rng = new_rng(42)
    items = [1, 2, 3, 4, 5]
    print("randint:", randint(rng, 10))
    shuffle(rng, items)
    print("shuffled:", items)

File: go/templates/rng.go
// go/templates/rng.go
package main

import (
	"fmt"
	"math/rand"
)

func main() {
	rng := rand.New(rand.NewSource(42))
	items := []int{1, 2, 3, 4, 5}
	fmt.Println("randint:", rng.Intn(10))
	rng.Shuffle(len(items), func(i, j int) {
		items[i], items[j] = items[j], items[i]
	})
	fmt.Println("shuffled:", items)
}
